### PR TITLE
Move golang packages under /pkg as is convention

### PR DESCRIPTION
**Pull request by benjaminapetersen**
Moves packages under `/pkg` as is convention for golang:
- auth
- version
- server

Top level `/pkg`, `/cmd` & `/vendor`.

/assign @spadgett 

Tested in my dev cluster, built new image & deployed, `go build` & `docker build` are still happy.

### CODE REVIEW

File: pkg/server/server.go [hunk 1]
```diff
@@ -14,9 +14,9 @@ import (
 	"github.com/coreos/pkg/capnslog"
 	"github.com/coreos/pkg/health"
 
-	"github.com/openshift/console/auth"
+	"github.com/openshift/console/pkg/auth"
 	"github.com/openshift/console/pkg/proxy"
-	"github.com/openshift/console/version"
+	"github.com/openshift/console/pkg/version"
```
Comment (spadgett) [inline pkg/server/server.go]: We need to update at least the `build-backend.sh` script if we move this.

Comment (spadgett) [reply pkg/server/server.go]: Also `test-backend.sh`. We should sweep the scripts to see if there are others this would cause problems for.

Comment (benjaminapetersen) [reply pkg/server/server.go]: Updating:

```bash
LD_FLAGS="-w -X github.com/openshift/console/version.Version=${GIT_TAG}"
LD_FLAGS="-w -X github.com/openshift/console/pkg/version.Version=${GIT_TAG}"
```

Comment (benjaminapetersen) [reply pkg/server/server.go]: Updated.
That file is all of `var Version string`, heh.

**Comment by benjaminapetersen** [comment]
```
Running tests...
can't load package: package github.com/openshift/console/auth: cannot find package "github.com/openshift/console/auth" in any of:
	/usr/local/go/src/github.com/openshift/console/auth (from $GOROOT)
	/go/src/github.com/openshift/console/auth (from $GOPATH)
can't load package: package github.com/openshift/console/server: cannot find package "github.com/openshift/console/server" in any of:
	/usr/local/go/src/github.com/openshift/console/server (from $GOROOT)
	/go/src/github.com/openshift/console/server (from $GOPATH)
```
Updated `test-backend.sh`

File: test-backend.sh [hunk 2]
```diff
@@ -17,7 +17,7 @@ export GOBIN=${PWD}/bin:${GOBIN}
 # Invoke ./cover for HTML output
 COVER=${COVER:-"-cover"}
 
-TESTABLE="auth pkg/proxy server"
+TESTABLE="pkg/auth pkg/proxy pkg/server"
 FORMATTABLE="${TESTABLE} cmd/bridge version"
```
Comment (spadgett) [inline test-backend.sh]:
```suggestion
FORMATTABLE="${TESTABLE} cmd/bridge pkg/version"
```

Comment (benjaminapetersen) [reply test-backend.sh]: got that! bash. bleh.

**Comment by benjaminapetersen** [comment]
passed.

**Pull request merged**